src/index.ts

```diff
@@ -99,50 +102,25 @@ function generateFakeFiscalCode(): FiscalCode {
   ) as FiscalCode;
 }
 
-function createDatabase(databaseName: string): Promise<Either<Error, void>> {
-  return new Promise((resolve) => {
-    documentClient.createDatabase({ id: databaseName }, (err, _) => {
-      if (err) {
-        return resolve(left<Error, void>(new Error(err.body)));
-      }
-      resolve(right<Error, void>(void 0));
-    });
-  });
-}
+const createDatabase = (databaseName: string): TE.TaskEither<Error, DatabaseResponse> =>
+  TE.tryCatch(
+    () =>
+      dbClient.databases.createIfNotExists({ id: databaseName }),
+    (err) =>
+      new Error(`Cannot create database ${databaseName}: ${err}`)
+  )
 
-function createCollection(
-  collectionName: string,
-  partitionKey: string
-): Promise<Either<Error, CollectionMeta>> {
-  return new Promise((resolve) => {
-    const dbUri = UriFactory.createDatabaseUri(cosmosDbName);
-    documentClient.createCollection(
-      dbUri,
-      {
-        id: collectionName,
-        partitionKey: {
-          kind: "Hash",
```

Comment:
what does this `kind` parameter do? i couldn't find it in the new library API. Is it implicit or did the key change?
@BurnedMarshal @michaeldisaro

Reply:
cosmos uses the hashed partitionkey by default as stated [here](https://docs.microsoft.com/it-it/azure/cosmos-db/large-partition-keys); so the library must do it automatically since it's not required anymore as a parameter.

old ref that shows the `Hash` kind was the only one: [link](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/f50b60511b53a42d47257d75d3f57bf4cee20d47/types/documentdb/index.d.ts#L227)